core/math/color.h

```diff
@@ -37,10 +37,10 @@
 struct Color {
 	union {
 		struct {
-			float r;
-			float g;
-			float b;
-			float a;
+			float r = 0.0;
+			float g = 0.0;
+			float b = 0.0;
+			float a = 0.0;
```

Comment:
`clang-tidy` obviously doesn't have a clue about what it's doing with unions...

I guess the best short-term fix is to undo all the changes it's trying to do in unions.

Is there a "proper" way to in-class initialize a union, especially with non-zero data like here? (poke @hpvb)